bundler/spec/support/builders.rb

```diff
@@ -138,6 +138,19 @@ def build_repo1
           s.write "lib/platform_specific.rb", "PLATFORM_SPECIFIC = '1.0.0 x86-darwin-100'"
         end
 
+        build_gem "darwin_single_arch" do |s|
+          s.platform = "ruby"
+          s.write "lib/darwin_single_arch.rb", "DARWIN_SINGLE_ARCH = '1.0 RUBY'"
+        end
+        build_gem "darwin_single_arch" do |s|
+          s.platform = "arm64-darwin"
+          s.write "lib/darwin_single_arch.rb", "DARWIN_SINGLE_ARCH = '1.0 arm64-darwin'"
+        end
+        build_gem "darwin_single_arch" do |s|
+          s.platform = "x86_64-darwin"
+          s.write "lib/darwin_single_arch.rb", "DARWIN_SINGLE_ARCH = '1.0 x86_64-darwin'"
+        end
+
```

Comment:
Sorry to bother you again @Bo98, just one more request. Since this gem is only used in the particular tests added here, I think it makes the tests more easily readable to move the creation of the dummy gem there. What I normally do is using an empty repository with only those gems, like this:

```ruby
build_repo4 do
  # definitions of the dummy gems here
end
```

And then use `file_uri_for(gem_repo4)` in test Gemfile and lock file.